android/authentication.md

```diff
@@ -1512,6 +1512,167 @@ If you want to sign out locally by just deleting tokens, you can call `signOut`
 AWSMobileClient.getInstance().signOut();
 ```
 
+## Customizing Authentication Flow
+
+Amazon Cognito User Pools supports customizing the authentication flow to enable custom challenge types, in addition to a password in order to verify the identity of users. These challenge types may include CAPTCHAs or dynamic challenge questions.
+
+To define your challenges for custom authentication flow, you need to implement three Lambda triggers for Amazon Cognito.
+
+For more information about working with Lambda Triggers for custom authentication challenges, please visit [Amazon Cognito Developer Documentation](https://docs.aws.amazon.com/cognito/latest/developerguide/user-pool-lambda-challenge.html).
+{: .callout .callout--info}
+
+### Custom Authentication in Amplify
+
+To initiate a custom authentication flow in your app, specify `authenticationFlowType` as `CUSTOM_AUTH` and  call `signIn` with a dummy password. A custom challenge needs to be answered using the `confirmSignnIn` method as follows:
+
+```java
+// Sample awsconfiguration.json
+{
+  "CognitoUserPool": {
+    "Default": {
+      "PoolId": "XX-XXXX-X_abcd1234",
+      "AppClientId": "XXXXXXXX",
+      "AppClientSecret": "XXXXXXXXX",
+      "Region": "XX-XXXX-X"
+    }
+  },
+  "Auth": {
+  "Default": {
+    "authenticationFlowType": "CUSTOM_AUTH"
+  }
+}
+}
+
+public void signIn() {
+    AWSMobileClient.getInstance().signIn(username, password, null, new Callback<SignInResult>() {
+        @Override
+        public void onResult(final SignInResult signInResult) {
+            runOnUiThread(new Runnable() {
+                @Override
+                public void run() {
+                    Log.d("APP", "Sign-in callback state: " + signInResult.getSignInState());
+                    switch (signInResult.getSignInState()) {
+                        case DONE:
+                            Log.d(TAG, "Sign-in done.");
+                            break;
+                        case SMS_MFA:
+                            Log.d(TAG, "Please confirm sign-in with SMS.");
+                            break;
+                        case NEW_PASSWORD_REQUIRED:
+                            Log.d(TAG, "Please confirm sign-in with new password.");
+                            break;
+                        case CUSTOM_CHALLENGE:
+                            confirmSignIn();
+                            break;
+                        default:
+                            Log.d(TAG, "Unsupported sign-in confirmation: " + signInResult.getSignInState());
+                            break;
+                    }
+                }
+            });
+        }
+
+        @Override
+        public void onError(Exception e) {
+            Log.e(TAG, "Sign-in error", e);
+        }
+    });
+}
+
+public void confirmSignIn() {
+    Map<String, String> res = new HashMap<String, String>();
+    res.put(CognitoServiceConstants.CHLG_RESP_ANSWER, "1133");
```

Comment:
I assume "1133" is your dummy result? You specify "5" in the "Creating a CAPTCHA" section below.

1. Make the values consistent to make it easier for the developer to follow the flow
1. Add a comment indicating that this is where the developer would inject the response from their app user for the custom challenge